tests/ssl-opt.sh

```diff
@@ -13948,13 +13948,13 @@ requires_openssl_3_x
 requires_protocol_version tls12
 requires_certificate_authentication
 requires_config_enabled MBEDTLS_SSL_RENEGOTIATION
-run_test    "Handshake defragmentation with server-initiated renegotiation: len=300" \
-            "$O_NEXT_SRV -tls1_2 -split_send_frag 300 -legacy_renegotiation -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key" \
-            "$P_CLI debug_level=3 renegotiation=1 request_page=/reneg" \
+run_test    "Handshake defragmentation with server-initiated renegotiation: len=256" \
+            "$O_NEXT_SRV -tls1_2 -split_send_frag 256 -legacy_renegotiation -cert $DATA_FILES_PATH/server5.crt -key $DATA_FILES_PATH/server5.key" \
+            "$P_CLI debug_level=3 renegotiation=1 renego_delay=32 request_page=/reneg" \
```

Comment:
Please add a comment explaining why `renego_delay` is needed here, for example:

The client has a limit on the number of records received before renegotiation completes - beyond this limit it will consider that the renegotiation requested has not been accepted. The default value (16 records) is fine without fragmentation, but fragmentation results in a larger number of records, so we need to up the limit by using the renego_delay option with a larger value.

Btw, why not use -1 (no limit)? Seems more robust.

Reply:
Good question. On the server side the -1 is set as part of the program, whereas on the library side, the default is `16`. The question is how much highlighting of this parameters we should do for our users?

I can and will push a new commit that will:
- Change the default behaviour of our program to always set it to -1 unless specified by the command line
- remove the renego_delay=32 from the ssl-opt.sh completely.
- Remove the comment added.

But that would hide valuable information about a use-case that we had to look closely to find out?

Reply:
I was not suggesting to change the program's default but rather to use -1 on the command line (instead of 32) for those test cases.

Reply:
I think it would be good to have the comment explaining that a higher `renego_delay` is needed, as documentation in case someone (us or a user) runs into this issue later. And if we have a comment, it might as well actually do something. So I lean towards an explicit higher `renego_delay` with a comment.

Reply:
I think we're in agreement about explicit, but rather than a somewhat abitrary number like 32, I'd prefer -1 which means unlimited. I'm really just suggesting `s/32/-1/` and nothing else.

Reply:
Otherwise tests with fragment size 256 will need 32, and if we add tests with fragment size 64 perhaps they'll need more, and with fragment size 4 even more... We're not testing the renego enforcment here, and it's getting in the way, so let's just disable it instead of guessing acceptable value is all I'm saying.